src/gpu/intel/jit/dsl/dsl.cpp

```diff
@@ -89,7 +107,7 @@ struct ctx_t {
             bool force_alloc = false) {
         auto type = type_t(
                 _type.kind(), _type.elems(), _type.attr() | type_attr_t::mut);
-        auto alloc_var = var(type, name);
+        auto alloc_var = var(type, ctx_->create_tmp_name(name));
```

Comment:
This looks wrong, why is it here?

Reply:
This is to avoid name duplicates in nested DSL, e.g.

```c++
void func() {
    auto tmp = def("tmp", ...);
    foo();
}

void foo() {
    auto tmp = def("tmp", ...);
}
```

In other words this is to track names and adjust to make names unique.

Reply:
I mainly dislike the divergence from normal languages design, so would prefer to avoid if it can be done in a simple and robust manner.
 
> If you suggest to add begin_scope()/end_scope() manually into every such a function - I don't think it's a good idea as this makes DSL usage harder 

What do you think of switching to an RAII pattern:
```
void foo() {
    auto guard = dsl::scope_guard();
    ....
}
```

Reply:
>  Do you see any particular issues?

I don't see any obvious issues and we can go ahead with this design, but I suspect this will lead to weird interactions in the future. It is just unclear to me if they will be legitimately problematic. For example, I could easily see us injecting subroutines rather than generating them in place to avoid redundant work. If we did that, we would get different IR between those methods which seems like it could expose issues.

Reply:
@rjoursler I agree, this is not perfect. I'm just thinking this way: what happens more often: either we create a new function/split existing while developing a kernel or we compare two IR outputs. My experience tells me that the former takes more time. Another observation (from my side): IR (being simpler and more stable) itself is less often at fault comparing with assembly. I can recall ~a dozen of issues that I spotted comparing assembly vs maybe a couple by comparing IR.

Reply:
Thanks, implemented both suggestions.